src/commands/node.mjs

```diff
@@ -2316,8 +2362,1051 @@ export class NodeCommand extends BaseCommand {
+           * @property {Map<String, NetworkNodeServices>} serviceMap
+           * @property {string} stagingDir
+           * @property {string} stagingKeysDir
+           * @property {PrivateKey} treasuryKey
+           * -- methods --
+           * @property {getUnusedConfigs} getUnusedConfigs
+           */
+          /**
+           * @callback getUnusedConfigs
+           * @returns {string[]}
+           */
+
+          // create a config object for subsequent steps
+          const config = /** @type {NodeUpdateConfigClass} **/ this.getConfig(NodeCommand.UPDATE_CONFIGS_NAME, NodeCommand.UPDATE_FLAGS_LIST,
+            [
+              'allNodeIds',
+              'existingNodeIds',
+              'freezeAdminPrivateKey',
+              'keysDir',
+              'nodeClient',
+              'podNames',
+              'serviceMap',
+              'stagingDir',
+              'stagingKeysDir',
+              'treasuryKey'
+            ])
+
+          config.curDate = new Date()
+          config.existingNodeIds = []
+
+          await self.initializeSetup(config, self.k8)
+
+          // set config in the context for later tasks to use
+          ctx.config = config
+
+          ctx.config.chartPath = await self.prepareChartPath(ctx.config.chartDirectory,
+            constants.FULLSTACK_TESTING_CHART, constants.FULLSTACK_DEPLOYMENT_CHART)
+
+          // initialize Node Client with existing network nodes prior to adding the new node which isn't functioning, yet
+          ctx.config.nodeClient = await this.accountManager.loadNodeClient(ctx.config.namespace)
+
+          const accountKeys = await this.accountManager.getAccountKeysFromSecret(FREEZE_ADMIN_ACCOUNT, config.namespace)
+          config.freezeAdminPrivateKey = accountKeys.privateKey
+
+          const treasuryAccount = await this.accountManager.getTreasuryAccountKeys(config.namespace)
+          const treasuryAccountPrivateKey = treasuryAccount.privateKey
+          config.treasuryKey = PrivateKey.fromStringED25519(treasuryAccountPrivateKey)
+
+          self.logger.debug('Initialized config', { config })
+        }
+      },
+      {
+        title: 'Identify existing network nodes',
+        task: async (ctx, task) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+          config.serviceMap = await self.accountManager.getNodeServiceMap(
+            config.namespace)
+          for (/** @type {NetworkNodeServices} **/ const networkNodeServices of config.serviceMap.values()) {
+            config.existingNodeIds.push(networkNodeServices.nodeName)
+          }
+
+          return self.taskCheckNetworkNodePods(ctx, task, config.existingNodeIds)
+        }
+      },
+      {
+        title: 'Load signing key certificate',
+        task: async (ctx, task) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+          const signingCertFile = Templates.renderGossipPemPublicKeyFile(constants.SIGNING_KEY_PREFIX, config.nodeId)
```

Comment:
if the flag is set for passing in the gossip signing key, generate the public key from the private key, and copy them into the cache/keys dir.  then load that.

Reply:
Updated

Reply:
As I'm thinking through this, I think we need the user to pass in both the public and the private for the TLS and/or gossip keys.

If it is TLS, we could generate a self-signed, but one with a CA would be better and they could do that and pass it in.

similar with the Gossip PEM key.  Then we wouldn't have to generate it.